Assets/Scripts/UI/JoinLobbyButton.cs

```diff
@@ -0,0 +1,16 @@
+using Lobby;
+using UnityEngine;
+
+namespace UI
+{
+    public class JoinLobbyButton : MonoBehaviour, IButtonPressHandler
```

Comment:
I'd argue this script doesn't really need to exist; we can add the relevant function(s) to the button's OnClick event instead. Whether we want to do this from the LobbyManager or from somewhere else at a surface level in our hierarchy, we generally want to avoid putting functionality directly onto our buttons, be it through adding Components to them, manually adding functions to the OnClick event in the inspector, or both. There is a time and a place for it, but it is generally best avoided.